src/prefect/utilities/executors.py

```diff
@@ -19,12 +20,29 @@
 StateList = Union["State", List["State"]]
 
 
-class Heartbeat(threading.Timer):
-    def run(self) -> None:
-        self.finished.wait(self.interval)  # type: ignore
-        while not self.finished.is_set():  # type: ignore
-            self.function(*self.args, **self.kwargs)  # type: ignore
-            self.finished.wait(self.interval)  # type: ignore
+class Heartbeat:
```

Comment:
Could you add a docstring or two to this class?

Reply:
@joshmeek UPDATED